### PR TITLE
VACMS-19227: Adds Philippines as a US state for facility purposes

### PR DESCRIPTION
**Pull request by omahane**
## Description

Relates to #19227 

## Testing done
Manually
Cypress (regression)

### Set up QA Content Publisher user
- [Log in](https://pr20068-moskbpwcbkbvekzpmwdfd0h8ke87xhpz.ci.cms.va.gov/) as an admin
- Assign the following roles to QA Content Publisher
  - [x] Content creator - VAMC
  - [x] Content publisher
- Assign the following Section to QA Content Publisher 
  - [x] VISN 21

### Add Philippines as a state to facilities
- As an admin, edit [Manila VA Clinic](https://pr20068-moskbpwcbkbvekzpmwdfd0h8ke87xhpz.ci.cms.va.gov/node/1059/edit)
- [x] Choose "Philippines" under **State**
- [x] Save the node as published
- [x] Edit [VA Regional Benefit Office at U.S. Embassy in Manila](https://pr20068-moskbpwcbkbvekzpmwdfd0h8ke87xhpz.ci.cms.va.gov/node/4298/edit)
- [x] Save as draft 

### 
### Create an event
- Log in as QA Content Publisher
- Make an event in the VA Manila Clinic **Section**
- Add an address for the location that is in the Philippines
- Save the node
- [x] Confirm that it saves without error
- [x] Confirm that the state is identified as "PH"

### Set the location for facilities
- Log in as an admin
- Go to the Manila VA Clinic
- Set the state as "Philippines"
- Save the node
- [x] Confirm that it saves without error
- [x] Confirm that the state is identified as "PH"
- Go to the VA Regional Benefit Office at U.S. Embassy in Manila
- Set the state as "Philippines"
- Save the node
- [x] Confirm that it saves without error
- [x] Confirm that the state is identified as "PH"

### Create a news release
- Log in as QA Content Publisher
- Create a new release
- [x] Confirm that Philippines is available as a **State**
- Fill out all required fields
- Save the release as draft
- [x] Confirm that the node saves without error
- [x] Confirm that "PH" is present in the saved release on node:view

### Edit a pre-existing news release
- Go to the [list of content](https://pr20068-moskbpwcbkbvekzpmwdfd0h8ke87xhpz.ci.cms.va.gov/section/vha/visn-21) in your section.
- Filter for a news release
- Edit the release
- [x] Confirm that the address is still in the US
- Save the node
- [x] Confirm that the node saves without error
- [x] Confirm that the US is present in the saved release on node:view

## Regression tests
- Creating content with a non-Philippines state
  - Cypress
    - [ ] Event
    - [ ] Press release

### Select Team for PR review

- [x] `CMS Team`
- [ ] `Public websites`
- [x] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### CODE REVIEW

**Comment by omahane** [comment]
@edmund-dunn @dsasser I have a change in this PR that replaces the default states in the US with a list that includes the Philippines, as we have facilities in the Philippines that are mapped to US addresses by way of the Lighthouse API: 
Screenshot 2024-12-10 at 9 02 00 AM

Do you think it makes more sense to keep it in the `va_gov_manila` module because it's only necessary because of the Manila facilities? Or do you think that the scope of the change makes it necessary to add to a module that encompasses more features, such as `va_gov_facilities` or `va_gov_backend`?

**Comment by omahane** [comment]
Based on the conversation on Slack, I chose to create a new module and add the code to that: https://dsva.slack.com/archives/CT4GZBM8F/p1733844761259909

File: docroot/modules/custom/va_gov_address/src/EventSubscriber/AddPhilippinesAsStateSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+
+  /**
+   * Provides the states of the US (plus the Philippines).
+   *
+   * @param \Drupal\address\Event\SubdivisionsEvent $event
+   *   The subdivisions event.
+   */
+  public function onSubdivisions(SubdivisionsEvent $event) {
+    $parents = $event->getParents();
+    if ($event->getParents() != ['US']) {
```
Comment (dsasser) [inline docroot/modules/custom/va_gov_address/src/EventSubscriber/AddPhilippinesAsStateSubscriber.php:30]: SHOULD:
I would suggest using strict comparison here, if possible. eg: `!==`

File: docroot/modules/custom/va_gov_address/src/EventSubscriber/AddPhilippinesAsStateSubscriber.php [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+    return $events;
+  }
+
+  /**
+   * Provides the states of the US (plus the Philippines).
+   *
+   * @param \Drupal\address\Event\SubdivisionsEvent $event
+   *   The subdivisions event.
+   */
+  public function onSubdivisions(SubdivisionsEvent $event) {
```
Comment (dsasser) [inline docroot/modules/custom/va_gov_address/src/EventSubscriber/AddPhilippinesAsStateSubscriber.php:28]: SHOULD:
This should probably have unit test coverage.

File: docroot/modules/custom/va_gov_address/src/EventSubscriber/AddPhilippinesAsStateSubscriber.php [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+    $parents = $event->getParents();
+    if ($event->getParents() != ['US']) {
+      return;
+    }
+
+    $definitions = [
+      'country_code' => $parents[0],
+      'parents' => $parents,
+      'subdivisions' => [
+        'AL' => [
```
[the remark this answers is not in the document]
Comment (omahane) [reply docroot/modules/custom/va_gov_address/src/EventSubscriber/AddPhilippinesAsStateSubscriber.php:38]: How would I get the Philippines in there, then?

[the remark this answers is not in the document]
Comment (dsasser) [reply docroot/modules/custom/va_gov_address/src/EventSubscriber/AddPhilippinesAsStateSubscriber.php:38]: You pull in the provided json and then add in the Philippines. Something like...

```php
// psuedo code
$definitions = // get definitions from .json
$definitions['subdivisions']['PH'] = [
  'code' => 'PH',
  'name' => 'Philippines',
  'country_code' => 'US',
  'id' => 'PH',
];
```

File: composer.lock [hunk 4]
```diff
@@ -4,7 +4,7 @@
         "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#installing-dependencies",
         "This file is @generated automatically"
     ],
-    "content-hash": "bce2825f1f2b4f9b1d1384930d11d322",
+    "content-hash": "aa992d810ef45398cacf8ee73a2a1bd9",
```
Comment (dsasser) [inline composer.lock:7]: Usually we don't see content-build changes showing up in PRs. Is this expected?

File: config/sync/core.extension.yml [hunk 5]
```diff
@@ -48,8 +48,6 @@ module:
   csv_serialization: 0
   ctools: 0
   ctools_block: 0
-  danse: 0
```
Comment (dsasser) [inline config/sync/core.extension.yml:51]: Looks like some unexpected changes leaked into your PR.

File: config/sync/metatag.metatag_defaults.node__event.yml [hunk 6]
```diff
@@ -5,7 +5,7 @@ dependencies: {  }
 id: node__event
 label: 'Content: Event'
 tags:
-  title: '[node:field_listing:entity:field_office] | [node:title] | Veterans Affairs'
+  title: '[node:title] | [node:field_listing:entity:field_office]  | Veterans Affairs'
```
Comment (dsasser) [inline config/sync/metatag.metatag_defaults.node__event.yml:8]: More unexpected changes.

File: docroot/modules/custom/va_gov_db/va_gov_db.install [hunk 7]
```diff
@@ -1166,3 +1166,20 @@ function va_gov_db_update_9015(&$sandbox) {
     'va_gov_github',
   ]);
 }
+
```
Comment (dsasser) [inline docroot/modules/custom/va_gov_db/va_gov_db.install:1169]: More unexpected changes.

File: docroot/themes/custom/vagovclaro/assets/js/anchor_links.js [hunk 8]
```diff
@@ -14,7 +14,8 @@
 
   Drupal.behaviors.vagovclaroAnchorLinks = {
     attach: function attach() {
-      $('a[href^="#"][href!="#"]:not([href^="#edit-group"])').click(function (event) {
+      //removed XOR operator because it was causing an issue where anchor lins were undefined VACMS-19375
```
Comment (dsasser) [inline docroot/themes/custom/vagovclaro/assets/js/anchor_links.js:17]: More unexpected changes.

**Review by dsasser — CHANGES_REQUESTED** [review]
Looks like possibly a rebase may have caused some things to get added to this PR that are not expected. See other comments.

**Review by dsasser — APPROVED** [review]
Great work. Code changes are awesome and the QA steps passed. Philippines is now available for all addresses across the CMS!

Screenshot 2024-12-18 at 5 38 11 PM
Screenshot 2024-12-18 at 5 29 05 PM

**Pull request merged**